lib/src/sql/model.rs

```diff
@@ -74,6 +78,35 @@ impl fmt::Display for Model {
 	}
 }
 
+impl Serialize for Model {
+	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+	where
+		S: serde::Serializer,
+	{
+		if is_internal_serialization() {
+			match self {
+				Self::Count(tb, c) => {
+					let mut serializer =
+						serializer.serialize_tuple_variant(TOKEN, 0, "Count", 2)?;
+					serializer.serialize_field(tb)?;
+					serializer.serialize_field(c)?;
+					serializer.end()
+				}
+				Self::Range(tb, b, e) => {
+					let mut serializer =
+						serializer.serialize_tuple_variant(TOKEN, 1, "Range", 3)?;
+					serializer.serialize_field(tb)?;
+					serializer.serialize_field(b)?;
+					serializer.serialize_field(e)?;
+					serializer.end()
+				}
+			}
+		} else {
+			serializer.serialize_some(&self.to_string())
```

Comment:
We should never get to serialising this type as a basic JSON/CBOR output, as we will always have called `compute` on this value by this time, and converted it to a primary type (Null, False, True, Number, Strand, Duration, Datetime, Uuid, Array, Object, Geometry). Therefore maybe a `serialize_none()` is better here?